File: backend/app/models/feedback.py
from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base


class Feedback(Base):
    __tablename__ = "feedback"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="SET NULL"), nullable=True)
    category = Column(String(50), nullable=False)  # bug, feature, general
    content = Column(Text, nullable=False)
    status = Column(String(20), default="open")  # open, in_progress, resolved
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    user = relationship("User", back_populates="feedback")

File: frontend/src/api/feedback.ts
import client from './client';

export interface Feedback {
  id: number;
  user_id?: number;
  category: string;
  content: string;
  status: string;
  created_at: string;
}

export interface CreateFeedbackRequest {
  category: string;
  content: string;
}

export const feedbackApi = {
  submit: (data: CreateFeedbackRequest) => client.post<Feedback>('/feedback', data),
  getMine: () => client.get<Feedback[]>('/feedback/mine'),
};
